### PR TITLE
Add SmolLM

### PR DESCRIPTION
**Pull request by loubnabnl**

### CODE REVIEW

File: smollm.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+The table below shows the performance of SmolLM-Instruct and other models on the IFEval benchmark (Prompt Strict Accuracy). Qwen2-1.5B-Instruct model scores the highest with 29.94, SmolLM-Instruct models provide a good balance between model size and performance, using only publicly available permissive datasets.
+
+<p align="center">
+ <img src="https://huggingface.co/datasets/HuggingFaceTB/images/resolve/main/Untitled%2016.png" alt=""  style="width: 60%; height: auto;"><br>
+<em>Evaluation of SmolLM-Instruct models on IFEval.</em>
+</p>
+
+# How to run locally ?
+
+Our models are designed to be small and can run locally on various hardware configurations. For reference, an iPhone 15 has 6GB of DRAM, while an iPhone 15 Pro has 16GB. These memory requirements make our models suitable for deployment on a wide range of devices, from smartphones to laptops. We benchmarked the memory footprint of our three model sizes:
```
Comment (lvwerra) [inline smollm.md:212]: no code example?

File: smollm.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+ <img src="https://huggingface.co/datasets/HuggingFaceTB/images/resolve/main/Untitled%208.png" alt=""  style="width: 90%; height: auto;"><br>
+<em>Comparison of Python-Edu to unfiltered Python code.</em>
+</p>
+
+# Training
+
+SmolLM models are available in three sizes and were trained on the data mixture below:
+
+- 130M and 360M models, each trained on 600B tokens from [Smollm-Corpus](https://huggingface.co/datasets/HuggingFaceTB/smollm-corpus)
+- 1.7B model, trained on 1T tokens from Smollm-Corpus
```
Comment (eliebak) [inline smollm.md:141]: SmoLM-Corpus instead of Smollm-Corpus

File: smollm.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+<p align="center">
+ <img src="https://huggingface.co/datasets/HuggingFaceTB/images/resolve/main/Untitled%208.png" alt=""  style="width: 90%; height: auto;"><br>
+<em>Comparison of Python-Edu to unfiltered Python code.</em>
+</p>
+
+# Training
+
+SmolLM models are available in three sizes and were trained on the data mixture below:
+
+- 130M and 360M models, each trained on 600B tokens from [Smollm-Corpus](https://huggingface.co/datasets/HuggingFaceTB/smollm-corpus)
```
Comment (eliebak) [inline smollm.md:140]: 135M instead of 130M

File: smollm.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+# How to run locally ?
+
+Our models are designed to be small and can run locally on various hardware configurations. For reference, an iPhone 15 has 6GB of DRAM, while an iPhone 15 Pro has 16GB. These memory requirements make our models suitable for deployment on a wide range of devices, from smartphones to laptops. We benchmarked the memory footprint of our three model sizes:
+
+<p align="center">
+ <img src="https://huggingface.co/datasets/HuggingFaceTB/images/resolve/main/Untitled%2013.png" alt=""  style="width: 60%; height: auto;"><br>
+<em>Memory footprint of SmolLM models.</em>
+</p>
+
+Along with the transformers checkpoints, we released ONNX checkpoints and plan to add a GGUF version compatible with `llama.cpp`. You can find a WebGPU demo of `HuggingFaceTB/SmolLM-135M-Instruct` here that shows how you can use these small models in your browser.
```
Comment (eliebak) [inline smollm.md:219]: supress the "here" i think and add links (tho it's just bellow so maybe no need for link

**Review by lvwerra — APPROVED** [review]

**Pull request merged**